docs/manual/determinism-checker.tex

```diff
@@ -152,6 +152,30 @@
   \label{fig-determinism-collections}
 \end{figure}
 
+\subsectionAndLabel{Accessing array elements}{determinism-access-array-elements}
```

Comment:
If the index used to access the array is nondeterministic, then the type of the array access should also be nondeterministic.  For example:
```
    void method(@NonDet int i, @Det String @Det [] array) {
        // :: error: (assignment.type.incompatible)
        @Det String s = array[i];
    }
```
The above code currently does not issue an error.  You could fix this in a separate pull request, but this section in the manual needs to include this, so it might be easy to do it in this pull request.